packages/client/src/generation/TSClient/common.ts

```diff
@@ -322,7 +322,7 @@ type Prisma__Pick<T, K extends keyof T> = {
 };
 
 
-export type Enumerable<T> = T | Array<T>;
+export type Enumerable<T> = T | ReadonlyArray<T>;
```

Comment:
I beleive that would unfortinately be a breaking change. Consider following pattern:

```ts
const input: UserWhereInput = { AND: [] };
input.AND.push(condition1)
input.AND.push(condition2)
```

This is the code that works for current version, but will break after PR is merged. Ideally, we should find a way to accept readonly arrays as input, while still allowing to build query input dynamically.